File: models/bookmark.go
package models

import "time"

type Bookmark struct {
	ID        int       `json:"id" gorm:"primaryKey"`
	UserID    int       `json:"user_id" gorm:"not null"`
	PhotoID   int       `json:"photo_id" gorm:"not null"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
	User      User      `json:"user"`
	Photo     Photo     `json:"photo"`
}

File: web/haiwangram/src/interface/Bookmark.ts
import Post from "./Post";

interface Bookmark {
  id: number;
  photo_id: number;
  user_id: number;
}

export interface BookmarkWithRelation {
  id: number;
  photo_id: number;
  user_id: number;
  photo: Post;
  // user: User;
}

export default Bookmark;
